test/basictests.jl

```diff
@@ -9,37 +9,52 @@ const Dual64 = ForwardDiff.Dual{Nothing, Float64, 1}
 n = 8
 A = Matrix(I, n, n)
 b = ones(n)
+# Real-valued systems
 A1 = A / 1;
 b1 = rand(n);
 x1 = zero(b);
+# A2 is similar to A1; created to test cache reuse
 A2 = A / 2;
 b2 = rand(n);
 x2 = zero(b);
+# Complex systems + mismatch types with eltype(tol)
+A3 = (A .+ rand(n) .* im) .|> ComplexF32;
+b3 = rand(n) .|> ComplexF32;
+x3 = zero(b) .|> ComplexF32;
+# A4 is similar to A3; created to test cache reuse
+A4 = (A .+ rand(n) .* im) .|> ComplexF32;
+b4 = b3 / 4 .|> ComplexF32;
+x4 = zero(b) .|> ComplexF32;
 
 prob1 = LinearProblem(A1, b1; u0 = x1)
 prob2 = LinearProblem(A2, b2; u0 = x2)
+prob3 = LinearProblem(A3, b3; u0 = x3)
+prob4 = LinearProblem(A4, b4; u0 = x4)
 
 cache_kwargs = (; verbose = true, abstol = 1e-8, reltol = 1e-8, maxiter = 30)
 
 function test_interface(alg, prob1, prob2)
-    A1 = prob1.A
-    b1 = prob1.b
-    x1 = prob1.u0
-    A2 = prob2.A
-    b2 = prob2.b
-    x2 = prob2.u0
+    A1, b1 = prob1.A, prob1.b
+    A2, b2 = prob2.A, prob2.b
```

Comment:
@ChrisRackauckas Not sure if this goes against SciML style, so feel free to revert back

Reply:
that's fine